docs/access_control/SQL_Access_Control_Design.md

```diff
@@ -0,0 +1,78 @@
+# Intro
+
+This document introduces the data structures used to store the issued tokens and the information associated with them.
+
+The type of tokens used are "Reference Tokens", it means they don't include any claims or data structure in themselves.
+The client sends the reference when authenticating a request, the recipient of the reference (API) must open a back-channel with the store and check the validity and claims of the token associated.
+
+
+# Token Table
+
+For the reasons mentioned before, the tokens must be stored on a table.
+
+The tokens in the table are indexed by the SHA-256 of the token reference.
+
+E.g. token reference
+```
+jXn2r5u8x/A%D*G-KaPdSgVkYp3s6v9y 
+```
+
+E.g. SHA-256 of token reference
+```
+06baeb46d42da63a97fd8358d5831714e95c826efe6f8d51e660457b3cb88185
+```
+
+(Hexadecimal representation, therefore, double the size of the token reference)
+
+Using SHA-256 to index the token reference instead of plaintext has the advantage that in case of a database leak the Attacker has no knowledge of the token reference (irreversible property) and can't make changes to the database.
+
+E.g. structure of token table:
+```
+CREATE TABLE dbo.Token(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    hash CHAR(64) UNIQUE, -- 64 hexa chars * 4 bits per char = 256 bits hash
+    isValid BOOLEAN,
+    scope_id INT REFERENCES dbo.scopes(id)
+)
+```
+
+# Claims Structures
+
+Associated to the token there can be a set of claims that provide additional information.
+e.g. issued at, expire time, subject...
+
+The token claims structures:
+```
+CREATE TABLE dbo.iat (
```

Comment:
I don't understand why we need tables for this. I recommend a "non-normalized" design, with a single table containing the following columns
- The token verification information (e.g. the token hash).
- Expiration
- Revocation status
- Other attributes such as creation data and last change data.
- JSON attribute with the claims, namely the scopes.

Reply:
I opted for including the columns creation data, expiration... in the JSON column, because https://www.iana.org/assignments/jwt/jwt.xhtml defines them as claims.

In case this is a wrong choice, I can correct it.